lib/CdeContext.tsx

```diff
@@ -1,83 +1,159 @@
-    setMapping: (mapping: DatasetCDEMapping) => void;
+
+    name: string;
+    datasetSample: DatasetSample;
+    datasetMapping: DatasetMapping;
+    datasetMappingHeader: string[];
+    collections: Collection[]
+    config: Config
+
+
+    // UI Module
     step: number;
     setStep: (step: number) => void;
     loadingMessage: string | null;
     setLoadingMessage: (loadingMessage: string | null) => void;
     errorMessage: string | null;
     setErrorMessage: (errorMessage: string | null) => void;
-    isOpen: boolean;
     handleClose: () => void;
-    infoOpen: boolean;
-    setInfoOpen: (open: boolean) => void;
 }>({
-    labName: '',
+
+    name: '',
+    datasetSample: [],
+    datasetMapping: {},
+    datasetMappingHeader: [],
+    collections: [],
     config: {
         width: "100%",
         height: "100%",
     },
-    inputMappings: [],
-    mapping: {} as DatasetCDEMapping,
-    setMapping: () => {},
+
     step: 0,
-    setStep: () => {},
+    setStep: () => {
+    },
     loadingMessage: null,
-    setLoadingMessage: () => {},
+    setLoadingMessage: () => {
+    },
     errorMessage: null,
-    setErrorMessage: () => {},
-    isOpen: true,
-    handleClose: () => { },
-    infoOpen: false,
-    setInfoOpen: () => {}
+    setErrorMessage: () => {
+    },
+    handleClose: () => {
+    },
 });
 
 export const useCdeContext = () => useContext(CdeContext);
 
-export const CdeContextProvider = ({children, labName, callback, inputMappings, config}: PropsWithChildren<InitParams>) => {
-    const [mapping, setMapping] = useState<DatasetCDEMapping>({});
-    const [step, setStep] = useState<number>(STEPS.HOME);
-    const [loadingMessage, setLoadingMessage] = useState<string | null>(null);
-    const [errorMessage, setErrorMessage] = useState<string | null>(null)
-    const [isOpen, setIsOpen] = useState<boolean>(true)
-    const [infoOpen, setInfoOpen] = useState<boolean>(false)
+export const CdeContextProvider = ({
+                                       datasetSample: rawDatasetSample,
+                                       datasetMapping: rawDatasetMapping,
+                                       additionalDatasetMappings: rawAdditionalDatasetMappings,
+                                       collections,
+                                       config,
+                                       name,
+                                       callback,
+                                       children
+                                   }: PropsWithChildren<InitParams>) => {
+
+    let datasetSample;
+    // Process and validate datasetSample
+    try {
+        validateDatasetSample(rawDatasetSample);
+        datasetSample = mapDatasetSample(rawDatasetSample);
+    } catch (error) {
+        const message = error instanceof Error ? error.message : 'An unknown error occurred';
+        // TODO: Check if it's fine to throw or should we show an error in the UI
+        throw new Error(`Invalid dataset sample: ${message}`);
```

Comment:
Should we address errors in the input ourselves or do we leave them to the host?

Reply:
the CDE dialog modal should at least report it in the console log. 

@ddelpiano we might also need an "Error" popup showing errors from the input files. I would say a "Nice to have"

Reply:
I did add support for snackbar errors, but this ones specifically I didn't know how to treat them because the solution would involve uploading new files, which is something that is done in the customer website (opposed to our modal).
Meanwhile I will add the console.error